Test dependencies should not be required at collection time

```
$ python -m pytest -k 'pyscf' test
==================================================================================================================================================================================== test session starts =====================================================================================================================================================================================
platform linux -- Python 3.11.6, pytest-7.4.4, pluggy-1.3.0
rootdir: /home/eric/development/cclib_berquist
configfile: pytest.ini
collected 148 items / 1 error / 148 deselected / 0 selected

=========================================================================================================================================================================================== ERRORS ===========================================================================================================================================================================================
____________________________________________________________________________________________________________________________________________________________________________ ERROR collecting test/test_bridge.py ____________________________________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/home/eric/development/cclib_berquist/test/test_bridge.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_bridge.py:20: in <module>
    from .bridge.testase import *
test/bridge/testase.py:19: in <module>
    raise ImportError("Must install ase to run this test")
E   ImportError: Must install ase to run this test
```

--- a/.gitignore
+++ b/.gitignore
@@ -14,3 +14,5 @@ venv
 .idea
 cclib/_version.py
 .direnv
+# Auto-generated from running tests, introduced by https://github.com/cclib/cclib/pull/1653
+dvb_gopt.xyz

--- a/test/bridge/testase.py
+++ b/test/bridge/testase.py
@@ -13,13 +13,7 @@ from cclib.bridge import cclib2ase
 from cclib.parser.utils import find_package
 
 import numpy as np
-
-if not find_package("ase"):
-    raise ImportError("Must install ase to run this test")
-
 import pytest
-from ase import Atoms
-from ase.calculators.emt import EMT
 
 if TYPE_CHECKING:
     from pathlib import Path
@@ -27,6 +21,10 @@ if TYPE_CHECKING:
 
 class ASETest:
     """Tests for the cclib2ase bridge in cclib."""
+
+    def setup_method(self) -> None:
+        if not find_package("ase"):
+            raise ImportError("Must install ase to run this test")
 
     def test_makease_allows_optimization(self) -> None:
         """Ensure makease works from direct input."""
@@ -44,6 +42,8 @@ class ASETest:
 
     def test_makecclib_retrieves_optimization(self) -> None:
         """Ensure makecclib works with native ASE Atoms objects."""
+        from ase import Atoms
+
         h2 = Atoms("H2", positions=[[0, 0, 0], [0, 0, 0.7]])
 
         # Check whether converting back gives the expected data,
@@ -86,6 +86,8 @@ class ASETest:
         assert np.isclose(ase_data.temperature, 0)
 
         # Make sure our object is compatible with ASE API.
+        from ase.calculators.emt import EMT
+
         dvb_sp_un.calc = EMT(label="dvb_sp_un")  # not a serious calculator!
 
         # Converting back should give updated results.

--- a/test/bridge/testchemfiles.py
+++ b/test/bridge/testchemfiles.py
@@ -6,7 +6,6 @@
 from cclib.bridge import cclib2chemfiles
 from cclib.parser.utils import find_package
 
-from chemfiles.frame import Frame
 from numpy.testing import assert_array_almost_equal
 
 from ..test_data import getdatafile
@@ -21,8 +20,10 @@ class ChemfilesTest:
 
         self.data, self.logfile = getdatafile("Gaussian", "basicGaussian16", ["dvb_sp.out"])
 
-    def test_makechemfiles(self):
+    def test_makechemfiles(self) -> None:
         """Check that geometry is transferred correctly from ccData to Frame"""
+        from chemfiles.frame import Frame
+
         frame = cclib2chemfiles.makechemfiles(self.data)
 
         assert isinstance(frame, Frame)
@@ -30,14 +31,14 @@ class ChemfilesTest:
 
         assert_array_almost_equal(frame.positions, self.data.atomcoords[-1])
 
-    def test_makechemfiles_masses(self):
+    def test_makechemfiles_masses(self) -> None:
         """Verify atomic masses are copied when present."""
         frame = cclib2chemfiles.makechemfiles(self.data)
         assert hasattr(self.data, "atommasses")
         frame_masses = [atom.mass for atom in frame.atoms]
         assert_array_almost_equal(frame_masses, self.data.atommasses)
 
-    def test_makechemfiles_charges(self):
+    def test_makechemfiles_charges(self) -> None:
         """Verify selected charge scheme is transferred correctly."""
         charges = getattr(self.data, "atomcharges", None)
 

